pkg/kclient/oc_server.go

```diff
@@ -60,7 +61,7 @@ func (c *Client) GetServerVersion(timeout time.Duration) (*ServerInfo, error) {
 
 	// fail fast if user is not connected (same logic as `oc whoami`)
 	_, err = c.userClient.Users().Get(context.TODO(), "~", metav1.GetOptions{})
-	if err != nil {
+	if err != nil && !kerrors.IsNotFound(err) {
```

Comment:
What if the user does not have permission to list users? The issue still persists:
```
❯ odo version -v 4
I0912 11:34:33.242229  223182 implem.go:106] The path for preference file is /home/asoro/.config/odo/preferences.yaml
I0912 11:34:33.242347  223182 segment.go:257] Checking telemetry enable status
I0912 11:34:33.242416  223182 util.go:733] HTTPGetRequest: https://raw.githubusercontent.com/redhat-developer/odo/main/build/VERSION
I0912 11:34:33.242515  223182 util.go:754] Response will be cached in /tmp/odohttpcache for 1h0m0s
I0912 11:34:33.244481  223182 oc_server.go:26] Trying to connect to server 10.10.10.105:6443
I0912 11:34:33.244638  223182 oc_server.go:33] Server https://10.10.10.105:6443 is up
I0912 11:34:33.249580  223182 version.go:77] an error occured while fetching server versionusers.user.openshift.io "~" is forbidden: User "user" cannot get resource "users" in API group "user.openshift.io" at the cluster scope
odo v3.0.0-rc1 (9a6c7ffa1)
I0912 11:34:33.249862  223182 odo.go:75] Could not get the latest release information in time. Never mind, exiting gracefully :)
```

This makes me wonder why we need to get all users. Maybe there are historical reasons behind this.. The comment reads "fail fast if user is not connected (same logic as `oc whoami`)", but does it really matter, as long as we are able to query the `/version/openshift` or `/version` API endpoints?

Reply:
>This makes me wonder why we need to get all users. Maybe there are historical reasons behind this.. The comment reads "fail fast if user is not connected (same logic as oc whoami)", but does it really matter, as long as we are able to query the /version/openshift or /version API endpoints?

I guess you're right. Maybe we don't need to query the user list. @dharmit @cdrage do you happen to know why do we have this fail fast?

Reply:
I'm honestly unsure with regards to why the fail test exists.